### DISCUSSION TITLE
Document how to format dates and numbers

### DISCUSSION THREAD

**Discussion by MarcSkovMadsen** [perspective-dev/perspective#2252] [Q&A] [answered]
Hi

I'm a contributor to HoloViz Panel and happy user of its `Perspective` *pane*. See https://panel.holoviz.org/reference/panes/Perspective.html

I would really like to understand and document how we can support and use date + number formatting. See https://github.com/holoviz/panel/issues/5066. In this issue I link to some indications on the `perspective` github and examples that it is in fact possible to format dates and numbers.

But I cannot find any clear documentation or examples on this.

It would be very helpful if you described date and number formatting in the docs and added a `formatting` example.

Just adding a few small examples to this issue would also help a lot right now.

## Additional Context

- We also support the [`Tabulator`](https://panel.holoviz.org/reference/widgets/Tabulator.html) pane. I would like the `Perspective` viewer Pane to be on par with this one with respect to formatting. You can see the formatting docs here for inspiration https://tabulator.info/docs/5.5/format

- I see other Perspective users not knowing how to format dates here https://github.com/finos/perspective/discussions/1442

**Comment by timkpaine** [comment, marked as answer]
Date and number formatting are both handled directly by the browser locale. 

e.g. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString

**Comment by MarcSkovMadsen** [comment]
Thanks for the answer.

As I understand this answer there is no built in or configurable way in Perspective to specify the formatting per column. The data needs to be preformatted?

If yes. The implication would be that we would have to format things "our selves" either on the Python or Javascript side? But then that could potentially destroy sorting for dates?

## Is there an example of data formatting?

I tried finding a Perspective example with Data and number formatting. But could not. Is there one. It would help me to have an example pinning out how this should be done.

## Is there an example of configuring the precission?

I can see in the streaming example that *precission* can be set manually. I know the background gradient just below can be configured in code. Is there an example configuring the *precission* in code?

![Precission](https://user-images.githubusercontent.com/42288570/244468244-405c4c9b-ba22-4931-8d36-7ccc031aa999.png)

**Reply by timkpaine** [reply]
https://github.com/finos/perspective/discussions/1458

**Comment by MarcSkovMadsen** [comment]
I would be really helpful to have a bullet list or table containing the configuration options for 

![image](https://github.com/finos/perspective/assets/42288570/b16947f8-9fff-4a66-a29e-61936ac24ecb)

somewhere.

I just discovered deep inside the file 

![image](https://github.com/finos/perspective/assets/42288570/8f853845-cb9b-449c-8899-480502620c89)

That I can see it using a `fixed`  value. First I had guessed on `precission` and `prec` without success.

**Comment by Laffery** [comment]
So how to format dates?

**Reply by texodus** [reply]
Click on the large `edit` button at the top of the date column.
